modules/Bio/EnsEMBL/Variation/Utils/VEP.pm

```diff
@@ -382,57 +384,76 @@ sub parse_line {
     return $vfs;
 }
 
-# sub-routine to detect format of input
-sub detect_format {
-    my $line = shift;
-    my @data = split /\s+/, $line;
+sub _valid_region_regex {
+  return qr/^([^:]+):(\d+)-(\d+)(:[-\+]?1)?[\/:]([a-z]{3,}|[ACGTN-]+)$/i;
+}
+
+# sub-routine to check format of string
+sub check_format {
+    my @line = @_;
+    my $format;
+
+    # any changes here must be copied to the JavaScript file to run instant VEP:
+    # public-plugins/tools/htdocs/components/20_VEPForm.js
+
+    # region: chr21:10-10:1/A
+    if ( scalar @line == 1 && $line[0] =~ &_valid_region_regex() ) {
+        $format = 'region';
+    }
 
     # SPDI: NC_000016.10:68684738:G:A
-    if (
-      scalar @data == 1 &&
-      $data[0] =~ /^(.*?\:){2}([^\:]+|)$/i
-    ) {
-      return 'spdi';
+    elsif ( scalar @line == 1 && $line[0] =~ /^(.*?\:){2}([^\:]+|)$/i ) {
+        $format = 'spdi';
+    }
+
+    # CAID: CA9985736
+    elsif ( scalar @line == 1 && $line[0] =~ /^CA\d{1,}$/i ) {
+        $format = 'caid';
     }
 
     # HGVS: ENST00000285667.3:c.1047_1048insC
     elsif (
-        scalar @data == 1 &&
-        $data[0] =~ /^([^\:]+)\:.*?([cgmrp]?)\.?([\*\-0-9]+.*)$/i
+        scalar @line == 1 &&
+        $line[0] =~ /^([^\:]+)\:.*?([cgmrp]?)\.?([\*\-0-9]+.*)$/i
     ) {
-        return 'hgvs';
+        $format = 'hgvs';
     }
 
     # variant identifier: rs123456
-    elsif (
-        scalar @data == 1
-    ) {
-        return 'id';
+    elsif ( scalar @line == 1 ) {
+        $format = 'id';
     }
 
     # VCF: 20  14370  rs6054257  G  A  29  0  NS=58;DP=258;AF=0.786;DB;H2  GT:GQ:DP:HQ
     elsif (
-        $data[0] =~ /(chr)?\w+/ &&
-        $data[1] =~ /^\d+$/ &&
-        $data[3] =~ /^[ACGTN\-\.]+$/i &&
-        $data[4] && $data[4] =~ /^([\.ACGTN\-\*]+\,?)+$|^(\<[\w]+\>)$/i
+        $line[0] =~ /(chr)?\w+/ &&
+        $line[1] =~ /^\d+$/ &&
+        $line[3] && $line[3] =~ /^[ACGTN\-\.]+$/i &&
+        $line[4]
```

Comment:
Why we are not checking regex for alternate allele?

Reply:
Hey @nakib103! The idea of this function is to check if this format is VCF: only these columns are needed to check that. I am simply copying here the code from ensembl-vep, already accepted in PR https://github.com/Ensembl/ensembl-vep/pull/1370.

Basically, we should not check for the alternative allele here because:
- The valid characters for the alternative allele are constantly changing with new versions of the [VCF specification](https://samtools.github.io/hts-specs/VCFv4.4.pdf)
- It only checks the first line (other lines may have issues and are ignored)
- The regex may not be up-to-date (it already happened that we support a specific type of alternative allele but forgot to update the regex here)

If the alternative allele is not supported, it should be the function [`create_VariationFeatures()` from `Parse::VCF`](https://github.com/Ensembl/ensembl-vep/blob/611114ef13d7bcbcfd330894711749c9a2030b04/modules/Bio/EnsEMBL/VEP/Parser/VCF.pm#L242) to return that issue with a warning (and not an error like `detect_format()` did, because the format is still VCF).

Reply:
Thanks Nuno,

That makes sense if we do not want to add checks to alt allele if its standard is changing frequently. But in that PR you still had checks for alts in the `detect_format()` -
https://github.com/nuno-agostinho/ensembl-vep/blob/7ecd5d059bb5cd396624804b8fdc0eb8a2f35334/modules/Bio/EnsEMBL/VEP/Parser.pm#L442

Is the change introduced here (not checking alts) intended?

Reply:
Thanks Nuno,

That makes sense if we do not want to add checks to alt allele if its standard is changing frequently. But in that PR you still had checks for alts in the detect_format() -
https://github.com/nuno-agostinho/ensembl-vep/blob/7ecd5d059bb5cd396624804b8fdc0eb8a2f35334/modules/Bio/EnsEMBL/VEP/Parser.pm#L442

Is the change introduced here (not checking alts) intended change for this PR?

Reply:
Yes, you are right, the PR still checks the alternative allele. Still, I think we should remove because of the reasons mentioned. I added this information (and the reasons why) in the PR description. Thanks!